### PR TITLE
Add Node 18 to tests, update to fetch-mock@9

### LINKED ISSUE #1052: "get Authenticated methods getItemResource defaults to read as blob" error in unit tests on Node v18
### Describe the bug

On Node v18, getting an error:

```
 get Authenticated methods getItemResource defaults to read as blob
  Message:
    Failed: response[readMethod] is not a function
```

This is happening for me locally and this is happening in GitHub Actions runners: https://github.com/Esri/arcgis-rest-js/pull/1048 (https://github.com/Esri/arcgis-rest-js/actions/runs/3735781808/jobs/6339403995)

### Reproduction

1. Install Node v18
2. `npm install`
3. `npm test`

### Logs

```shell
Failures:
1) get Authenticated methods getItemResource defaults to read as blob
  Message:
    Failed: response[readMethod] is not a function
  Stack:
    TypeError: response[readMethod] is not a function
        at file:///C:/.../arcgis-rest-js/packages/arcgis-rest-portal/src/items/get.ts:442:29
  Message:
    Error: Timeout - Async function did not complete within 5000ms (set by jasmine.DEFAULT_TIMEOUT_INTERVAL)
  Stack:
    Error: Timeout - Async function did not complete within 5000ms (set by jasmine.DEFAULT_TIMEOUT_INTERVAL)
        at <Jasmine>
        at listOnTimeout (node:internal/timers:564:17)
        at processTimers (node:internal/timers:507:7)

607 specs, 1 failure
Finished in 25.106 seconds
```

### System Info

```shell
Node: v18.12.0
NPM: v8.19.2
```

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by patrickarlt**
This PR fixes #1052 by upgrading from `fetch-mock@5.0.0` to `fetch-mock@9.0.0` and all the ugliness that produces. This mostly involved removing all our manually defined types and fixing a few key tests.

### CODE REVIEW

**Review by gavinr — APPROVED** [review]
I tested this locally (on Node v18.12.0), and the tests are now all passing! 💯

File: package.json [hunk 1]
```diff
@@ -72,17 +72,18 @@
     "@semantic-release/git": "^10.0.1",
     "@semantic-release/npm": "^8.0.0",
     "@skypack/package-check": "^0.2.2",
-    "@types/fetch-mock": "^6.0.6",
+    "@types/fetch-mock": "^7.3.5",
     "@types/jasmine": "^2.8.2",
     "@types/node": "^12.20.4",
     "@typescript-eslint/eslint-plugin": "^4.4.0",
     "@typescript-eslint/parser": "^4.31.0",
+    "babel-preset-env": "^1.7.0",
```
Comment (tomwayson) [inline package.json:80]: I don't see how `babel-preset-env` is used, did you mean to add this?

**Review by tomwayson — APPROVED** [review]
LGTM - one question

File: package.json (hunk 1 above)
Comment (patrickarlt) [reply package.json:80]: @tomwayson it is used in https://github.com/Esri/arcgis-rest-js/pull/1077/files#diff-251ba8e0b8d64329c3861fda004bbd931a1e2887551033edf97fbba22d57d36eR52 to configure Babel

**Pull request merged**